### ISSUE TITLE
Invisible columns show up after resetting state

### ISSUE THREAD

**Issue by antego** [wbstr/vaadin-filtertablestate#5]
Steps to reproduce:
1. Create container with to columns "column1" and "column2"
2. Create FilterTable and set only column1 as visible
3. Reset table's state
4. Now column2 is also present

[Full demo code](https://gist.github.com/antego/c2abebf51ca11f11ca2df72ce8c3c90e)

``` java
    private FilterTable buildFilterTable() {
        FilterTable table = new FilterTable();
        table.setSizeFull();
        table.setContainerDataSource(buildContainer());
        table.setFilterBarVisible(true);
        table.setColumnCollapsingAllowed(true);
        table.setColumnReorderingAllowed(true);

        //we want only "visible_column" to be shown
        table.setVisibleColumns("visible_column");

        return table;
    }

    private void addFilterTableExample() throws ServerRpcManager.RpcInvocationException {
        FilterTable filterTable = buildFilterTable();
        FilterTableState state = new FilterTableState();
        state.extend(filterTable, createFilterTableStateHandler());
        layout.addComponent(filterTable);

        Object[] initialVisibleColumns = filterTable.getVisibleColumns();

        Assert.assertArrayEquals(initialVisibleColumns, new Object[] {"visible_column"});

        ServerRpcManager<FilterTableStateRpc> rpc = (ServerRpcManager<FilterTableStateRpc>) state
                .getRpcManager("com.wcs.wcslib.vaadin.widget.filtertablestate.shared.FilterTableStateRpc");
        rpc.applyInvocation(new ServerRpcMethodInvocation("", FilterTableStateRpc.class, "resetProfile", 0));

        Object[] visibleColumnsAfterReset = filterTable.getVisibleColumns();

        //there it is, should be just "visible_column"
        Assert.assertArrayEquals(visibleColumnsAfterReset, new Object[] {"visible_column", "invisible_column"});
    }
```

**Comment by a-schild** [comment]
Any chance to get this published as a new 1.1.2 release in the vaadin addons?

**Comment by juger89** [comment]
Thank you for your cooperation. The fix is available in version 1.1.2.
